docroot/profiles/custom/cgov_site/themes/custom/ncids_trans/front-end/src/lib/components/nci-promo-block/cgdp-promo-block.ts

```diff
@@ -0,0 +1,72 @@
+import { landingClickTracker } from '../../core/analytics/landing-page-contents-helper';
+
+/**
+ * Gets the title of the PromoBlock
+ * @param {HTMLElement} anchorEl - Clicked anchor element.
+ */
+const getTitle = (element: HTMLElement): string =>
+	element.querySelector('.nci-promo-block__heading')?.textContent || '_ERROR_';
+
+/**
+ * Gets the theme of the PromoBlock (Light / Dark)
+ * @param {HTMLElement} element - The Promo Block
+ */
+const getPromoTheme = (element: HTMLElement): string =>
+	element.classList.contains('nci-promo-block--dark') ? 'Dark' : 'Light';
+
+/**
+ * Gets the variant of the PromoBlock (No Image / Image Left / Image Right)
+ * @param {HTMLElement} element - The Promo Block
+ */
+const getPromoVariant = (element: HTMLElement): string => {
+	if (element.classList.contains('nci-alternating-right')) {
+		return 'Image Right';
+	} else if (element.classList.contains('nci-promo-block--with-image')) {
+		return 'Image Left';
+	} else return 'No Image';
+};
+
+/**
+ * Click handler for the language toggle click.
+ * @param promoBlockRow the guide card row element.
+ */
+const promoBlockLinkClickHandler =
+	(promoBlock: HTMLElement) => (evt: Event) => {
+		evt.preventDefault();
+		evt.stopPropagation();
```

Comment:
you'll want to remove these - this was only added during development for testing

Reply:
Oops!